### PR TITLE
cbor field sorting for struct autogen

### PR DESCRIPTION
**Pull request by whyrusleeping**
This is still a WIP, but I think its heading in the right direction

### CODE REVIEW

**Comment by warpfork** [comment]
I think I'm mostly happy with this, but would not get *rid* of the ability to set key sorter modes per specific map type.

Also, when you rebase, could you be so kind as to do the `-s` signy thing?

File: obj/atlas/structMapAutogen.go [hunk 1]
```diff
@@ -8,21 +8,26 @@ import (
...
 	entry := &AtlasEntry{
 		Type:      rt,
-		StructMap: &StructMap{Fields: exploreFields(rt, tagName)},
+		StructMap: &StructMap{Fields: exploreFields(rt, tagName, sorter)},
 	}
 	return entry
 }
 
+const (
+	StructFieldSort_Default = "default"
```
Comment (warpfork) [inline obj/atlas/structMapAutogen.go]: Missing a type for the consts, plz.

Maybe this is actually the same type and enum when talking about either maps or structs?

File: obj/marshalMapWildcard.go [hunk 2]
```diff
@@ -47,13 +47,19 @@ func (mach *marshalMachineMapWildcard) Reset(slab *marshalSlab, rv reflect.Value
 		mach.keys[i].rv = v
 		mach.keys[i].s = v.String()
 	}
-	switch mach.cfg.MapMorphism.KeySortMode {
+
+	ksm := atlas.KeySortMode_Default
+	if mach.morphism != nil {
```
Comment (warpfork) [inline obj/marshalMapWildcard.go]: I don't think this branch should ever be reachable.  The marshaller *always* sets it during dispatch.

File: obj/marshalSlab.go [hunk 3]
```diff
@@ -118,8 +118,8 @@ func _yieldMarshalMachinePtr(row *marshalSlabRow, atl atlas.Atlas, rt reflect.Ty
 		case entry.StructMap != nil:
 			row.marshalMachineStructAtlas.cfg = entry
 			return &row.marshalMachineStructAtlas
-		case entry.MapMorphism != nil:
-			row.marshalMachineMapWildcard.cfg = entry
+		case atl.MapMorphism != nil:
+			row.marshalMachineMapWildcard.morphism = atl.MapMorphism
```
Comment (warpfork) [inline obj/marshalSlab.go]: This line should revert.  If there *is* an explicit entry in the atlas for this type, we should take it.  (People will probably write that *and* be setting the default at the whole atlas scale at the same time incredibly rarely, but still.)

File: obj/marshalMapWildcard.go (hunk 2 above)
Comment (whyrusleeping) [reply obj/marshalMapWildcard.go]: I don't think thats true... If i take this out i get panics in tests

**Comment by whyrusleeping** [comment]
@warpfork Alrighty. Hows that. I even signed the commit off like a good citizen

**Comment by warpfork** [comment]
whoooopwhoops testsss

**Pull request merged**